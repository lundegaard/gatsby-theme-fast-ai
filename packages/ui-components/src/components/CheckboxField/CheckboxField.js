import React from 'react';
import PropTypes from 'prop-types';

import Label from '../Label';
import Checkbox from '../Checkbox';
import Box from '../Box';
import SuperFieldHint from '../SuperFieldHint';

const getVariant = ({ hasError, readOnly, disabled }) => {
	if (hasError) {
		return 'error';
	}
	if (disabled || readOnly) {
		return 'disabled';
	}
};

const CheckboxField = ({ label, hasError, disabled, readOnly, hint, ...rest }) => (
	<Box variant={getVariant({ hasError, readOnly, disabled })} pb={2}>
		<Label fontSize={1} alignItems="center">
			<Checkbox {...rest} />
			{label}
		</Label>
		{hint ? <SuperFieldHint>{hint}</SuperFieldHint> : null}
	</Box>
);

CheckboxField.propTypes = {
	/** Disabled state */
	disabled: PropTypes.bool,
	/** Error state */
	hasError: PropTypes.bool,
	/** Hint itext */
	hint: PropTypes.node,
	/** Label for Checkbox  */
	label: PropTypes.node,
	/** Called when `onChange` event is triggered. */
	onChange: PropTypes.func,
	/** Read-only state */
	readOnly: PropTypes.bool,
	/** Current value of checkbox. */
	value: PropTypes.any,
};

export default CheckboxField;
