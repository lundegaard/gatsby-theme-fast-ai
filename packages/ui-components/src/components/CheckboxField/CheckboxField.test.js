import React from 'react';
import { shallow } from 'enzyme';

import CheckboxField from './CheckboxField';

describe('<CheckboxField />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<CheckboxField {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
