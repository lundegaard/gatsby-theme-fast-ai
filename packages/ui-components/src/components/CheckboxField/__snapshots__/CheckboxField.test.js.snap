// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxField /> matches snapshot when shallowly rendered 1`] = `
<Box
  pb={2}
>
  <Label
    alignItems="center"
    fontSize={1}
  >
    <Checkbox />
  </Label>
</Box>
`;
